crates/bevy_ecs/src/system/combinator.rs

```diff
@@ -421,8 +421,30 @@ where
         &mut self,
         world: UnsafeWorldCell,
     ) -> Result<(), SystemParamValidationError> {
-        // SAFETY: Delegate to other `System` implementations.
-        unsafe { self.a.validate_param_unsafe(world) }
+        // SAFETY: Delegate to the `System` implementation for `a`.
+        let a_validation = unsafe { self.a.validate_param_unsafe(world) };
+
+        // SAFETY: Delegate to the `System` implementation for `b`.
+        let b_validation = unsafe { self.b.validate_param_unsafe(world) };
+
+        match (a_validation, b_validation) {
+            (Ok(()), Ok(())) => Ok(()),
+            (Ok(_), Err(err_b)) => Err(err_b),
+            (Err(err_a), Ok(_)) => Err(err_a),
+            (Err(err_a), Err(err_b)) => {
+                // Combine the errors from both systems, using the most severe outcome.
+                // TODO: consider combining the error strings in a more intelligible way
+                // but be mindful of the possibility of multiple piped systems!
+                Err(SystemParamValidationError {
+                    // Only skip if both systems should be skipped.
+                    // Panicking systems should not be skipped!
+                    skipped: err_a.skipped && err_b.skipped,
+                    message: err_a.message + err_b.message,
```

Comment:
Should we add a test to check the message is formatted legibly (i.e. no weird spacing issues)? Or is that juice not worth the squeeze?

Reply:
Not worth the squeeze IMO: way too fragile. That said, I'll toss in some spaces, free of charge!